libraries/@mattermost/hardware-keyboard/src/index.tsx

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {useEffect} from 'react';
+import {NativeEventEmitter, NativeModules, Platform} from 'react-native';
+
+const LINKING_ERROR =
+  'The package \'mattermost-hardware-keyboard\' doesn\'t seem to be linked. Make sure: \n\n' +
+  Platform.select({ios: "- You have run 'pod install'\n", default: ''}) +
+  '- You rebuilt the app after installing the package\n' +
+  '- You are not using Expo Go\n';
+
+// @ts-expect-error global
+const isTurboModuleEnabled = global.__turboModuleProxy != null;
+
+const MattermostHardwareKeyboardModule = isTurboModuleEnabled ? require('./NativeMattermostHardwareKeyboard').default : NativeModules.MattermostHardwareKeyboard;
+
+const MattermostHardwareKeyboard = MattermostHardwareKeyboardModule || new Proxy(
+    {},
+    {
+        get() {
+            throw new Error(LINKING_ERROR);
+        },
+    },
+);
+
+const emitter = new NativeEventEmitter(MattermostHardwareKeyboard);
+
+type Event = {
+  action: string;
+}
+
+type Events = {
+  onEnterPressed?: () => void;
+  onShiftEnterPressed?: () => void;
+  onFindChannels?: () => void;
+}
+
+export function useHardwareKeyboardEvents(events: Events) {
```

Comment:
The way this is implemented, on every render of the component, this is removing the listener and adding it again, since the effect has no dependencies. I have no strong opinions against leaving this as is, but wanted to flag it in case calling add listener and remove too often have any undesirable effect.

Reply:
adding an empty array of dependencies should solve it..